encrypted-config-value/src/main/java/com/palantir/config/crypto/EncryptedValue.java

```diff
@@ -18,21 +18,32 @@
 
 import static com.google.common.base.Preconditions.checkArgument;
 
+import com.google.common.io.BaseEncoding;
 import com.palantir.config.crypto.algorithm.Algorithm;
 import com.palantir.config.crypto.algorithm.Algorithms;
 import java.io.IOException;
+import javax.annotation.Nullable;
 import org.immutables.value.Value;
 
 @Value.Immutable
 public abstract class EncryptedValue {
 
     private static final String PREFIX = "enc:";
 
-    public abstract String encryptedValue();
+    @Nullable
+    public abstract String getAlgorithm();
```

Comment:
though in reality I think we should always populate this field, when it's not found during parsing we can put the default value into it, which makes the remaining consumer code not have to branch